src/mapport.cpp

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) 2011-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <mapport.h>
+
+#include <clientversion.h>
+#include <logging.h>
+#include <net.h>
+#include <netaddress.h>
+#include <netbase.h>
+#include <threadinterrupt.h>
+#include <util/system.h>
+
+#ifdef USE_NATPMP
+#include <natpmp.h>
+#ifdef WIN32
+#include <winsock2.h>
```

Comment:
I think we can avoid this platform dependent include dance by including `<src/compat.h>`.

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/18077#issuecomment-748361576).